[python3] Build error

Package: python3[core]:x64-windows -> 3.10.7#1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.34.31933.0
-    vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
    vcpkg-scripts version: 62d01b70d 2022-11-25 (27 hours ago)

**To Reproduce**

`vcpkg install boost:x64-windows`

**Failure logs**

```
-- Using cached python-cpython-v3.10.7.tar.gz.
-- Cleaning sources at C:/Src/Libs/vcpkg/buildtrees/python3/src/v3.10.7-bd5f6dab3e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Src/Libs/vcpkg/downloads/python-cpython-v3.10.7.tar.gz
-- Applying patch 0001-only-build-required-projects.patch
-- Applying patch 0003-use-vcpkg-zlib.patch
-- Applying patch 0004-devendor-external-dependencies.patch
-- Applying patch 0005-dont-copy-vcruntime.patch
-- Applying patch 0008-python.pc.patch
-- Applying patch 0009-bz2d.patch
-- Applying patch 0010-dont-skip-rpath.patch
-- Applying patch 0007-workaround-windows-11-sdk-rc-compiler-error.patch
-- Using source at C:/Src/Libs/vcpkg/buildtrees/python3/src/v3.10.7-bd5f6dab3e.clean
-- Building PCbuild/pcbuild.proj for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: C:/Src/Libs/vcpkg/downloads/tools/powershell-core-7.3.0-windows/pwsh.exe -noprofile -executionpolicy Bypass -nologo -file C:/Src/Libs/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Src/Libs/vcpkg/packages/python3_x64-windows/tools/python3/_freeze_importlib.exe -installedDir C:/Src/Libs/vcpkg/packages/python3_x64-windows/bin -verbose
    Working Directory: C:/Src/Libs/vcpkg
    Error code: Access violation
    See logs for more information:
      C:\Src\Libs\vcpkg\buildtrees\python3\copy-tool-dependencies-0-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_copy_tool_dependencies.cmake:9 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_copy_tool_dependencies.cmake:41 (z_vcpkg_copy_tool_dependencies_search)
  scripts/cmake/vcpkg_install_msbuild.cmake:98 (vcpkg_copy_tool_dependencies)
  ports/python3/portfile.cmake:147 (vcpkg_install_msbuild)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\Src\Libs\vcpkg\buildtrees\python3\copy-tool-dependencies-0-err.log</summary>

```
Fatal error. 0xC0000005
   at System.Management.Automation.Security.SystemPolicy+WldpNativeMethods.WldpCanExecuteFile(System.Guid, WLDP_EXECUTION_EVALUATION_OPTIONS, IntPtr, System.String, WLDP_EXECUTION_POLICY ByRef)
   at System.Management.Automation.Security.SystemPolicy.GetFilePolicyEnforcement(System.String, System.IO.FileStream)
   at System.Management.Automation.ExternalScriptInfo.ReadScriptContents()
   at System.Management.Automation.Language.Parser.ParseFile(System.String, System.Management.Automation.Language.Token[] ByRef, System.Management.Automation.Language.ParseError[] ByRef)
   at Microsoft.PowerShell.ConsoleHost.DoRunspaceInitialization(Microsoft.PowerShell.RunspaceCreationEventArgs)
   at Microsoft.PowerShell.ConsoleHost.DoCreateRunspace(Microsoft.PowerShell.RunspaceCreationEventArgs)
   at Microsoft.PowerShell.ConsoleHost.CreateRunspace(Microsoft.PowerShell.RunspaceCreationEventArgs)
   at Microsoft.PowerShell.ConsoleHost.DoRunspaceLoop(System.String, Boolean, System.Collections.ObjectModel.Collection`1<System.Management.Automation.Runspaces.CommandParameter>, Boolean, System.String, System.String)
   at Microsoft.PowerShell.ConsoleHost.Run(Microsoft.PowerShell.CommandLineParameterParser, Boolean)
   at Microsoft.PowerShell.ConsoleHost.Start(System.String, System.String, Boolean)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(System.String[], Int32)
   at Microsoft.PowerShell.ManagedPSEntry.Main(System.String[])
```
</details>

**Additional context**

[copy-tool-dependencies-0-err.log](https://github.com/microsoft/vcpkg/files/10096347/copy-tool-dependencies-0-err.log)

I've updated the installed version of PowerShell core to 7.3.0

This is on a Windows 11 Home 22H2 PC. The same package versions built successfully on a Windows 10 Pro 21H2 PC.

--- a/scripts/azure-pipelines/windows/deploy-pwsh.ps1
+++ b/scripts/azure-pipelines/windows/deploy-pwsh.ps1
@@ -5,5 +5,5 @@
 
 # REPLACE WITH UTILITY-PREFIX.ps1
 
-$PwshUrl = 'https://github.com/PowerShell/PowerShell/releases/download/v7.3.0/PowerShell-7.3.0-win-x64.msi'
+$PwshUrl = 'https://github.com/PowerShell/PowerShell/releases/download/v7.2.7/PowerShell-7.2.7-win-x64.msi'
 InstallMSI -Url $PwshUrl -Name 'PowerShell Core'

--- a/scripts/vcpkgTools.xml
+++ b/scripts/vcpkgTools.xml
@@ -194,11 +194,11 @@
         <archiveName>ninja-freebsd-1.8.2.zip</archiveName>
     </tool>
     <tool name="powershell-core" os="windows">
-        <version>7.3.0</version>
+        <version>7.2.7</version>
         <exeRelativePath>pwsh.exe</exeRelativePath>
-        <url>https://github.com/PowerShell/PowerShell/releases/download/v7.3.0/PowerShell-7.3.0-win-x86.zip</url>
-        <sha512>da01f53bc8e3f11b99ad936c5925074ea379ff6aec9e9e4f6040e406de5fa08cd266f619913ce16aa7883b2c6a1e0e2bbf2c3521c0ec4feba2531ca2cc61f363</sha512>
-        <archiveName>PowerShell-7.3.0-win-x86.zip</archiveName>
+        <url>https://github.com/PowerShell/PowerShell/releases/download/v7.2.7/PowerShell-7.2.7-win-x86.zip</url>
+        <sha512>e17b307b66d17d4ece4d2f93080e6111265070b2ee6b77ffe3224973a470c59a14b1b17fab7445b18d370ac1720045188a9a011f0c9c8538ac56eca7673f184a</sha512>
+        <archiveName>PowerShell-7.2.7-win-x86.zip</archiveName>
     </tool>
     <tool name="node" os="windows">
         <version>16.15.1</version>

